tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs

```diff
@@ -358,6 +382,28 @@ private async void RestorePinnedWidgets(object sender, RoutedEventArgs e)
                     Log.Logger()?.ReportError("DashboardView", $"RestorePinnedWidgets(): ", ex);
                 }
             }
+
+            // Now that we've ordered the widgets, put them in their final collection.
+            var finalPlace = 0;
+            foreach (var orderedWidget in restoredWidgetsWithPosition)
+            {
+                var widget = orderedWidget.Value;
+                var size = await widget.GetSizeAsync();
+                await InsertWidgetInPinnedWidgetsAsync(widget, size, finalPlace);
+                await WidgetHelpers.SetPositionOnWidget(widget, finalPlace);

```

Comment:
Does `SetPositionOnWidget` need to be called after `InsertPineedWidgetAsync`?

Reply:
Yeah. Maybe I should rename to make it clearer that it's updating a text field on the widget, not changing the order in the list. We need to set it here because it's possible the position either a) wasn't set on this widget before or b) was set, but the widget got moved up in the order because other widgets went away, so we make sure the widget's state is updated with the potentially new position.